cmd/scanCluster.go

```diff
@@ -0,0 +1,116 @@
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"text/template"
+	"strings"
+
+	"github.com/jgsqware/clairctl/clair"
+	"github.com/jgsqware/clairctl/config"
+	"github.com/jgsqware/clairctl/docker"
+	"github.com/jgsqware/clairctl/xstrings"
+	"github.com/spf13/cobra"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/pkg/api/v1"
+	"github.com/spf13/viper"
+)
+
+var scanCmd = &cobra.Command{
+	Use:   "scanCluster",
+	Short: "Scan and analyze all Docker images in cluster",
+	Long:  `Scan and analyze all Docker images in cluster, against Ubuntu, Red hat and Debian vulnerabilities databases`,
+	Run: func(cmd *cobra.Command, args []string) {
+
+		clusterConfig, err := rest.InClusterConfig()
+		if err != nil {
+			log.Fatalf("Error scanCluster must be run from inside a cluster %v", err)
+		}
+		// creates the clientset
+		clientset, err := kubernetes.NewForConfig(clusterConfig)
+		if err != nil {
+			panic(err.Error())
+		}
+		pods, err := clientset.CoreV1().Pods("").List(v1.ListOptions{})
+		if err != nil {
+			panic(err.Error())
+		}
+		fmt.Printf("There are %d pods in the cluster\n", len(pods.Items))
+
+		images := make(map[string]struct{})
+
+		for _, pod := range pods.Items {
+			for _, container := range pod.Spec.Containers {
+				images[container.Image] = struct{}{}
+			}
+		}
+
+		fmt.Printf("list of images %v", images)
+
+		for imageName := range (images) {
+			config.ImageName = imageName
```

Comment:
I would extract Method from cmd/Analyze.go and replace line 52,72 with the method